### DISCUSSION TITLE
What's the recommended way to handle type errors?

### DISCUSSION THREAD

**Discussion by nigellima** [mobxjs/mobx-state-tree#1813] [General] [answered]
Hi, everyone.

So, I just experienced something running MST in prod mode. 
In dev mode, when something from the API does not correspond the types we defined in our models, it console.error the type error information. However, in production mode the error is never there, but still the models are not created due to the type inconsistency. It took me a while to figure out why some information was not being displayed but there was no error in the console.

Basically I was fetching the API information and then casting it to my observable, something like:

```
try {
   const results = yield FooService.fetchAll();
   self.foo = cast(results);
} catch (error)  {
    console.error(error)
}
```

In dev mode, the error is caught in runtime, but in prod it never reaches the catch statement. Is there a recommended way to handle these scenarios?

Thanks in advance

**Comment by k-g-a** [comment, marked as answer]
Hi! You could use [typecheck()](https://mobx-state-tree.js.org/API/#typecheck) helper function for this.
Alternatively, documentation states that you can set `process.env.ENABLE_TYPE_CHECK = "true"` if you want to enable type checking in any environment. A quick look into the code proves this statement.
